api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/rtt/ReturnedDateIITDateDaysCalculation.java

```diff
@@ -191,25 +198,185 @@ private CalculationResultMap getMostRecentDateFromPatientWhoExperincedITT(
             + "                            AND obs.value_datetime IS NOT NULL "
             + "                            AND enc.encounter_type = ${msterCardDrugPickupEncounterType}  "
             + "                            AND enc.location_id = :location "
-            + "                            AND obs.value_datetime <= :previousPeriod "
+            + "                            AND obs.value_datetime <= '"
+            + endDate
+            + "' "
             + "                       GROUP  BY pa.patient_id "
             + "                   ) most_recent "
             + "               GROUP BY most_recent.patient_id "
-            + "               HAVING final_encounter_date < :previousPeriod ";
+            + "               HAVING final_encounter_date < '"
+            + endDate
+            + "' ";
 
     StringSubstitutor stringSubstitutor1 = new StringSubstitutor(map1);
     String replacedQeury1 = stringSubstitutor1.replace(query1);
 
-    String finalQuery = replacedQeury1;
+    //
+
+    Map<String, Integer> map2 = new HashMap<>();
+    map2.put(
+        "adultoSeguimentoEncounterType",
+        hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId());
+    map2.put(
+        "ARVPediatriaSeguimentoEncounterType",
+        hivMetadata.getPediatriaSeguimentoEncounterType().getEncounterTypeId());
+    map2.put(
+        "aRVPharmaciaEncounterType",
+        hivMetadata.getARVPharmaciaEncounterType().getEncounterTypeId());
+    map2.put(
+        "masterCardDrugPickupEncounterType",
+        hivMetadata.getMasterCardDrugPickupEncounterType().getEncounterTypeId());
+    map2.put("returnVisitDateConcept", commonMetadata.getReturnVisitDateConcept().getConceptId());
+    map2.put(
+        "returnVisitDateForArvDrugConcept",
+        hivMetadata.getReturnVisitDateForArvDrugConcept().getConceptId());
+    map2.put("artDatePickup", hivMetadata.getArtDatePickupMasterCard().getConceptId());
+
+    String query2 =
+        " SELECT pat.patient_id,  eeecc.encounter_datetime   "
+            + " FROM   patient pat "
+            + " INNER JOIN encounter eeecc ON eeecc.encounter_id = pat.patient_id "
```

Comment:
When having table aliases, can we have something that will be closer to the object? for example renaming **eeecc** to **enc** maybe the first 3 latters of a relation can be used and if there could be several, we could differentiate them with numbers in addition to letters

Reply:
done